### PR TITLE
Change SCEP authority initialization

### PR DESCRIPTION
**Pull request by hslatman**

### CODE REVIEW

File: authority/options.go [hunk 1]
```diff
@@ -205,6 +206,23 @@ func WithX509SignerFunc(fn func() ([]*x509.Certificate, crypto.Signer, error)) O
...
 }
 
+// func WithSCEPOptions(crt *x509.Certificate, s crypto.Signer, d crypto.Decrypter) Option {
+// 	return func(a *Authority) error {
+// 		a.scepCertificate = crt
+// 		a.scepSigner = s
+// 		a.scepDecrypter = d
+// 		return nil
+// 	}
+// }
```
Comment (maraino) [inline authority/options.go:216]: Remove me.

File: authority/options.go [hunk 2]
```diff
@@ -205,6 +206,23 @@ func WithX509SignerFunc(fn func() ([]*x509.Certificate, crypto.Signer, error)) O
...
+
+func WithFullSCEPOptions(options *scep.Options) Option {
```
Comment (maraino) [inline authority/options.go:218]: Add comment and add something like:
```go
// This feature is EXPERIMENTAL and might change at any time.
```

File: authority/authority.go [hunk 3]
```diff
@@ -671,54 +674,57 @@ func (a *Authority) init() error {
...
+					// only pass the decrypter down when it was successfully created,
+					// meaning it's an RSA key, and `CreateDecrypter` did not fail.
+					options.Decrypter = decrypter
+					options.DecrypterCert = options.Intermediates[0]
+				}
+			}
+
+			// provide the current SCEP provisioner names, so that the provisioners
+			// can be validated when the CA is started.
+			options.SCEPProvisionerNames = a.getSCEPProvisionerNames()
```
Comment (maraino) [inline authority/authority.go:712]: I would set this to `scepOptions` too if not provided.

**Review by maraino — CHANGES_REQUESTED** [review]

File: authority/authority.go (hunk 3 above)
Comment (hslatman) [reply authority/authority.go]: Fixed in c0fbace88227fcfeb5f92415fa6158e4859638b3. Is that what you had in mind?

File: authority/options.go (hunk 2 above)
Comment (hslatman) [reply authority/options.go:212]: Fixed in c0fbace88227fcfeb5f92415fa6158e4859638b3.

File: authority/options.go (hunk 1 above)
Comment (hslatman) [reply authority/options.go]: Fixed in c0fbace88227fcfeb5f92415fa6158e4859638b3.

**Review by maraino — APPROVED** [review]

**Pull request merged**